### PR TITLE
support polymorphism in example generation

### PR DESCRIPTION
**Pull request by necusjz**

### CODE REVIEW

File: src/aaz_dev/swagger/controller/_example_builder.py [hunk 1]
```diff
@@ -76,11 +93,49 @@ def build(self, var_prefix, example_dict):
...
-                    item = ExampleItem(command=self.command, arg_var=f"{var_prefix}.{name}", key=name, val=value)
-
-                example_items += self.build(item.arg_var, value)
+                    item = ExampleItem(
+                        command=self.command,
+                        cmd_operation=self.cmd_operation,
+                        arg_var=f"{var_prefix}.{name}",
+                        key=name,
+                        val=value
+                    )
```
Comment (kairu-ms) [inline src/aaz_dev/swagger/controller/_example_builder.py]: It's better using a builder function, which will return None when the arg_var is invalid, else return the ExampleItem instance.

File: src/aaz_dev/swagger/controller/_example_builder.py [hunk 2]
```diff
@@ -76,11 +93,49 @@ def build(self, var_prefix, example_dict):
...
+                    disc_item = ExampleItem(
+                        command=self.command,
+                        arg_var=f"{item.arg_var}.{safe_value}"
+                    )
+
+                    formatted = dict()
+                    if disc_name := disc_item.arg_option:
+                        formatted[disc_name] = value
+                    else:
+                        formatted[safe_value] = value
```
Comment (kairu-ms) [inline src/aaz_dev/swagger/controller/_example_builder.py]: What's this used for?

Comment (kairu-ms) [reply src/aaz_dev/swagger/controller/_example_builder.py]: I think directly assign the value as formatted is not correct. You should use desc_item to build the formatted value.

File: src/aaz_dev/swagger/controller/_example_builder.py [hunk 3]
```diff
@@ -62,34 +97,104 @@ def mapping(self, example_dict):
...
+                safe_value = self.get_safe_value(disc.value)
+                disc_item = ExampleItem.new_instance(
+                    command=self.command,
+                    cmd_operation=self.cmd_operation,
+                    arg_var=f"{var_prefix}.{safe_value}"
+                )
+
+                if disc_item and (disc_name := disc_item.arg_option):
+                    example_obj[disc_name] = example_obj.copy()
+                    example_items += self.build(disc_item.arg_var, example_obj[disc_name])
```
Comment (kairu-ms) [inline src/aaz_dev/swagger/controller/_example_builder.py]: Can this work? Why need to copy and assign in example_obj?
```suggestion
                    example_items += self.build(disc_item.arg_var, example_obj.copy())
```

**Comment by kairu-ms** [comment]
The rest LGTM. Good job.

File: src/aaz_dev/swagger/controller/_example_builder.py (hunk 3 above)
Comment (necusjz) [reply src/aaz_dev/swagger/controller/_example_builder.py]: aim to continently trim the inner/outer layer, although copy is OK, deepcopy might be better

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**